bgpd/bgp_route.c

```diff
@@ -12519,7 +12704,67 @@ DEFPY (show_ip_bgp_instance_neighbor_advertised_route,
 	if (argv_find(argv, argc, "route-map", &idx))
 		rmap_name = argv[++idx]->arg;
 
-	return peer_adj_routes(vty, peer, afi, safi, type, rmap_name, uj, wide);
+	if (!all)
+		return peer_adj_routes(vty, peer, afi, safi, type, rmap_name,
+				       show_flags);
+
+	if (uj)
+		vty_out(vty, "{\n");
+
+	if (CHECK_FLAG(show_flags, BGP_SHOW_OPT_AFI_IP)
+	    || CHECK_FLAG(show_flags, BGP_SHOW_OPT_AFI_IP6)) {
+		afi = CHECK_FLAG(show_flags, BGP_SHOW_OPT_AFI_IP) ? AFI_IP
+								  : AFI_IP6;
+		FOREACH_SAFI (safi) {
+			if (strmatch(get_afi_safi_str(afi, safi, true),
+				     "Unknown"))
+				continue;
+
+			if (!bgp_afi_safi_peer_exists(bgp, afi, safi))
+				continue;
+
+			if (uj) {
+				if (first)
```

Comment:
can we rewrite this logic using just json-c functions?

Reply:
I mean without crafting JSON manually, it's a mess for future changes..

Reply:
addressed this comment. updated the sample json output in commit message. please check

Reply:
With the latest commit, Tests are failing due to mismatch in the JSON representation.
1. ERROR :  $->routes: d1 has element of type 'Array' but the corresponding element in d2 is of type 'Object'
    with the existing manual crafting, array is printed in { } instead of [ ]. 
    so with json_object_object_add(json, "routes", json_paths); 
     Router2# show ip bgp json 
     {
        "vrfId":0,
        "vrfName":"default",
        "tableVersion":4,
        "routerId":"128.16.16.1",
        "defaultLocPrf":100,
        "localAS":2,
        "routes":[
           "128.16.16.0\/24",
           [
              {

2. "128.16.16.0\/24", instead of "128:16:16.0/24": 
    with json_object_array_add(<array> , json_object_new_string(buf));

3. Memory leaks occur  
    bgp_show_table(json_object *json)
    {
                 if (rd) {
                        json_object_object_add(json_rd_val, rd, json_paths);
                        json_object_object_add(json_rd_desc, "routeDistinguishers", json_rd_val);
                        json_object_object_add(json, "routes", json_rd_desc);
                } else
                        json_object_object_add(json, "routes", json_paths);

                json_object_free(json_paths or json_rd_val or json_rd_desc);//freeing up the local objects causes crash.
                return CMD_SUCCESS;
    }

Reply:
@ton31337 
There are some changes (mentioned in the above comment) in the JSON format output when JSON-C APIs are used.
So tests are failing. 
Do we need to change the test scripts?